src/org/skypro/skyshop/article/SearchEngine.java

```diff
@@ -10,18 +10,16 @@
 
 public class SearchEngine {
     private List<Searchable> searchable = new LinkedList<>();
-    private int size = 0;
 
 
-    public List search(Searchable whatSearch) {
+
+    public LinkedList search(Searchable whatSearch) {
```

Comment:
Во-первых, нужно всегда оперировать интерфейсными ссылками, это тема полиморфизма. Во-вторых, типизацию тоже надо использовать
```suggestion
    public List<Searchable> search(Searchable whatSearch) {
```